cloudinit/distros/gentoo.py

```diff
@@ -31,18 +37,33 @@ def __init__(self, name, cfg, paths):
         self.osfamily = "gentoo"
         # Fix sshd restarts
         cfg["ssh_svcname"] = "/etc/init.d/sshd"
+        if distros.uses_systemd():
+            LOG.error("Cloud-init does not support systemd with gentoo")
```

Comment:
gentoo can use system (or not (another init)), I'm not sure how this function checks for systemd support.

Reply:
@prometheanfire - The reason I was considering adding that is because cloud-init doesn't currently have support for systemd with Gentoo. Cloud-init assumes openrc for gentoo. I'm not sure if the ebuild packaging allows cloud-init with systemd, but if so things may fail and this log may help confused users. Alternatively we could add support for systemd to cloud-init for gentoo. I don't think it would be much work to make that happen, but I don't know if anybody would find that useful.

Reply:
`distutils-r1_python_install --init-system=sysvinit_openrc,systemd --distro gentoo` is how we are installing cloud-init, so we are at least installing service files and the like already.  Systemd on Gentoo is fully supported, even if gentoo is better known for openrc (I personally use systemd on my laptop with gentoo).

Reply:
@prometheanfire: I think we may be misunderstanding each other in some way.

Gentoo fully supports systemd, but cloud-init unconditionally assumes that Gentoo's init system is OpenRC.

For example, note that the [init command here](https://github.com/canonical/cloud-init/blob/6a4b6dc0de94a9e75bb44fa6c03288c7d4dded7e/cloudinit/distros/gentoo.py#L23) is rc-service, and [`apply_locale()`](https://github.com/canonical/cloud-init/blob/6a4b6dc0de94a9e75bb44fa6c03288c7d4dded7e/cloudinit/distros/gentoo.py#L35) doesn't support the [systemd locale setup with `localectl`](https://wiki.gentoo.org/wiki/Localization/Guide/en#systemd_2).

I haven't tried using cloud-init with a systemd gentoo install, but since cloud-init assumes OpenRC-specific things, I don't think it  could start services correctly.

Reply:
@prometheanfire - Thanks for the feedback, it's much appreciated. Would you mind taking a another look at this considering my latest commit and comments above regarding systemd/cloud-init?  We have an upcoming release (22.1) and I'd like to get this in before the release if possible.